src/main/java/com/commercetools/project/sync/util/SyncUtils.java

```diff
@@ -46,13 +46,16 @@ public static String getApplicationVersion() {
       @Nonnull final String resourceName,
       @Nonnull final SyncException exception,
       @Nonnull final Optional<T> resource,
-      @Nonnull final List<UpdateAction<T>> updateActions) {
+      @Nullable final List<UpdateAction<T>> updateActions) {
```

Comment:
You are making `@Nonnull` annotation to `@Nullable` and wrote 2 test cases for it. But in our source code, I believe all these time we never passed `null` to `updateActions`. I guess it would be valid case to leave the annotation as `@Nonnull` and pass an emptyList to `updateActions`. Then assert whether we get the appropriate log message printed.

Reply:
Actually we do pass null, that's why I made these changes 😅 To be more even accurate, this null is coming from java-sync in error callback and it comes here.